src/main/java/com/rybka/service/ExchangeService.java

```diff
@@ -1,100 +1,73 @@
     }
 
-    public void exchange(Double amount) {
-
-        this.amount = amount;
-        this.total = amount * currencyValue;
-    }
-
-    public void getCurrencyObject() {
-        Currency currency = new Currency(baseCurrencyAbbreviation, targetCurrencyAbbreviation, currencyValue, amount, total);
-        HibernateDAO hibernateDAO = new HibernateDAO();
-        hibernateDAO.save(currency);
-        hibernateDAO.showTableRow();
-    }
-
-    public Double getCurrencyValue() {
-        return currencyValue;
-    }
-
-    public void setCurrencyValue(Double currencyValue) {
-        this.currencyValue = currencyValue;
-    }
-
-    public String getBaseCurrencyAbbreviation() {
-        return baseCurrencyAbbreviation;
+    private Currency constructDataObject(Double currencyRate, String userBaseCurrencyAbbreviation,
+                                         String userTargetCurrencyAbbreviation, Double amount) {
+        currency.setCurrencyValue(currencyRate);
+        currency.setBaseCurrencyAbbreviation(userBaseCurrencyAbbreviation);
+        currency.setTargetCurrencyAbbreviation(userTargetCurrencyAbbreviation);
+        currency.setAmount(amount);
+        currency.setTotal(currency.getAmount() * currency.getCurrencyValue());
+        return currency;
     }
 
-    public void setBaseCurrencyAbbreviation(String baseCurrencyAbbreviation) {
-        this.baseCurrencyAbbreviation = baseCurrencyAbbreviation;
-    }
-
-    public String getTargetCurrencyAbbreviation() {
-        return targetCurrencyAbbreviation;
-    }
-
-    public void setTargetCurrencyAbbreviation(String targetCurrencyAbbreviation) {
-        this.targetCurrencyAbbreviation = targetCurrencyAbbreviation;
-    }
-
-    public Double getAmount() {
-        return amount;
-    }
-
-    public void setAmount(Double amount) {
-        this.amount = amount;
-    }
-
-    public Double getTotal() {
-        return total;
+    private HttpResponse<String> sendRequest(String userBaseCurrencyAbbreviation) {
+        String url = String.format("https://prime.exchangerate-api.com/v5/%s/latest/%s",
+                API_KEY, userBaseCurrencyAbbreviation);
+        try {
+            HttpClient client = HttpClient.newHttpClient();
+            HttpRequest request = HttpRequest.newBuilder()
+                    .uri(URI.create(url))
+                    .build();
+
+            return client.send(request,
+                    HttpResponse.BodyHandlers.ofString());
+        } catch (InterruptedException | IOException interruptedException) {
+            interruptedException.printStackTrace();
+        }
+        return null;
     }
 
-    public void setTotal(Double total) {
-        this.total = total;
+    private Double retrieveJsonData(HttpResponse<String> response, String userTargetCurrencyAbbreviation) {
+        ObjectMapper mapper = new ObjectMapper();
+        var currencyRate = 0.0;
+        try {
+            JsonNode rootNode = mapper.readTree(response.body());
```

Comment:
Try to deserialize response content and using java stream api find target currency.

Reply:
fixed